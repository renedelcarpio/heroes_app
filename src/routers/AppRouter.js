import { BrowserRouter, Route, Routes } from 'react-router-dom';

import DcScreen from '../components/dc/DcScreen';
import LoginScreen from '../components/login/LoginScreen';
import MarvelScreen from '../components/marvel/MarvelScreen';
import { Navbar } from '../components/ui/Navbar';
import React from 'react';
import SearchScreen from '../components/search/SearchScreen';

const AppRouter = () => {
	return (
		<BrowserRouter>
			<Navbar />
			<Routes>
				<Route path='/marvel' element={<MarvelScreen />} />
				<Route path='/dc' element={<DcScreen />} />
				<Route path='/search' element={<SearchScreen />} />
				<Route path='/login' element={<LoginScreen />} />
			</Routes>
		</BrowserRouter>
	);
};

export default AppRouter;
