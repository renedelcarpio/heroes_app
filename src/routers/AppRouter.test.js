import AppRouter from './AppRouter';
import { AuthContext } from '../auth/authContext';
import { mount } from 'enzyme';

describe('Tests in AppRouter component', () => {
	test('should show login if user is not authenticated', () => {
		const contextValue = {
			user: {
				logged: false,
			},
		};
		const wrapper = mount(
			<AuthContext.Provider value={contextValue}>
				<AppRouter />
			</AuthContext.Provider>
		);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('h1').text().trim()).toBe('Login');
	});

	test('should show main page if user is authenticated', () => {
		const contextValue = {
			user: {
				logged: true,
				name: 'userName',
			},
		};
		const wrapper = mount(
			<AuthContext.Provider value={contextValue}>
				<AppRouter />
			</AuthContext.Provider>
		);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('.navbar').exists()).toBe(true);
	});
});
