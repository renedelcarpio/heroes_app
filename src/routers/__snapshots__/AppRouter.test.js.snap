// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in AppRouter component should show login if user is not authenticated 1`] = `
<div
  className="login"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="button-login"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;

exports[`Tests in AppRouter component should show main page if user is authenticated 1`] = `
Array [
  <nav
    className="navbar"
  >
    <a
      className="navbar__brand"
      href="/"
      onClick={[Function]}
    >
      Heroes App
    </a>
    <div
      className="navbar__heroes"
    >
      <div
        className="navbar__nav"
      >
        <a
          className="navbar__nav-item "
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="navbar__nav-item "
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="navbar__nav-item "
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar__login"
    >
      <ul
        className="navbar__nav name-button"
      >
        <span
          className="navbar__nav-item user-name"
        >
          userName
        </span>
        <button
          className="navbar__nav-item button"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="marvel__container"
  >
    <div
      className="marvel__image-container"
    >
      <img
        alt="DC Logo"
        className="marvel__image"
        src="https://thecouch.world/wp-content/uploads/2018/02/Marvel-Comics-Logo-Desktop-Wallpaperconv.jpg"
      />
    </div>
    <div
      className="marvel__cards-container"
    >
      <div
        className="card__container"
      >
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Spider Man"
              height={300}
              src="/assets/marvel-spider.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Spider Man
            </h2>
            <h3>
              Peter Parker
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-spider"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Captain America"
              height={300}
              src="/assets/marvel-captain.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Captain America
            </h2>
            <h3>
              Steve Rogers
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-captain"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Iron Man"
              height={300}
              src="/assets/marvel-iron.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Iron Man
            </h2>
            <h3>
              Tony Stark
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-iron"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Thor"
              height={300}
              src="/assets/marvel-thor.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Thor
            </h2>
            <h3>
              Thor Odinson
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-thor"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Hulk"
              height={300}
              src="/assets/marvel-hulk.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Hulk
            </h2>
            <h3>
              Bruce Banner
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-hulk"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Wolverine"
              height={300}
              src="/assets/marvel-wolverine.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Wolverine
            </h2>
            <h3>
              James Howlett
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-wolverine"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Daredevil"
              height={300}
              src="/assets/marvel-daredevil.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Daredevil
            </h2>
            <h3>
              Matthew Michael Murdock
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-daredevil"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Hawkeye"
              height={300}
              src="/assets/marvel-hawkeye.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Hawkeye
            </h2>
            <h3>
              Clinton Francis Barton
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-hawkeye"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Cyclops"
              height={300}
              src="/assets/marvel-cyclops.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Cyclops
            </h2>
            <h3>
              Scott Summers
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-cyclops"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Silver Surfer"
              height={300}
              src="/assets/marvel-silver.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Silver Surfer
            </h2>
            <h3>
              Norrin Radd
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-silver"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
